src/osx/carbon/font.cpp

```diff
@@ -949,86 +951,153 @@ bool wxNativeFontInfo::FromString(const wxString& s)
+            return false;
+        m_encoding = (wxFontEncoding)l;
+        return true;
+    }
+    else if ( version == 2 )
+    {
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        bool underlined = l != 0;
 
-    return true;
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        bool strikethrough = l != 0;
+
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        wxFontEncoding encoding = (wxFontEncoding)l;
+
+        wxString xml = tokenizer.GetString();
+        if ( !xml.StartsWith("<?xml"))
+            xml+=s_plistPrefix;
+        wxCFStringRef plist(xml);
+        wxCFDataRef listData(CFStringCreateExternalRepresentation(kCFAllocatorDefault,plist,kCFStringEncodingUTF8,0));
+        wxCFDictionaryRef attributes((CFDictionaryRef) CFPropertyListCreateWithData(kCFAllocatorDefault, listData, 0, NULL, NULL));
+        CTFontDescriptorRef descriptor = NULL;
+        if (attributes != NULL)
+            descriptor = CTFontDescriptorCreateWithAttributes(attributes);
+        if (descriptor != NULL)
+        {
+            InitFromFontDescriptor(descriptor);
+            m_underlined = underlined;
+            m_strikethrough = strikethrough;
+            m_encoding = encoding;
+            return true;
+        }
+    }
+
+    return false;
 }
 
 wxString wxNativeFontInfo::ToString() const
 {
     wxString s;
 
-    s.Printf(wxT("%d;%s;%d;%d;%d;%d;%d;%s;%d"),
-        1, // version
-        wxString::FromCDouble(GetFractionalPointSize()),
-        GetFamily(),
-        (int)GetStyle(),
-        GetNumericWeight(),
-        GetUnderlined(),
-        GetStrikethrough(),
-        GetPostScriptName().GetData(),
-        (int)GetEncoding());
+    // version 2 is a streamed property list of the font descriptor as recommended by Apple
+    // prefixed by the attributes that are non-native to the native font ref like underline, strikethrough etc.
+    wxCFDictionaryRef attributes(CTFontDescriptorCopyAttributes(GetCTFontDescriptor()));
+
+    if (attributes != NULL)
+    {
+        CFPropertyListFormat format = kCFPropertyListXMLFormat_v1_0;
+        if (CFPropertyListIsValid(attributes, format))
+        {
+            wxCFDataRef listData(CFPropertyListCreateData(kCFAllocatorDefault, attributes, format, 0, NULL));
+            wxCFStringRef cfString( CFStringCreateFromExternalRepresentation( kCFAllocatorDefault, listData, kCFStringEncodingUTF8) );
+            wxString xml = cfString.AsString();
+            xml.Replace("\r",wxEmptyString,true);
+            if ( xml.StartsWith(s_plistPrefix ))
+                xml = xml.Mid(s_plistPrefix.length());
+
+            s.Printf(wxT("%d;%d;%d;%d;"),
+                 2, // version
+                 GetUnderlined(),
+                 GetStrikethrough(),
+                 (int)GetEncoding());
+
+            s += xml;
```

Comment:
Trivial, but as long we're using printf, why not construct the entire string using it, i.e.
```suggestion
            s.Printf("%d;%d;%d;%d;%s",
                 2, // version
                 GetUnderlined(),
                 GetStrikethrough(),
                 (int)GetEncoding(),
                 xml);
```